File: src/node/magma/app/compile/type/ArrayType.ts
import { Type } from "../../../../magma/api/Type";
export class ArrayType implements Type {
	child: Type;
	constructor (child: Type) {
		this.child = child;
	}
	generate(): string {
		return this/*auto*/.child.generate(/*auto*/) + "[]";
	}
	isFunctional(): boolean {
		return false/*auto*/;
	}
	isVar(): boolean {
		return false/*auto*/;
	}
	generateBeforeName(): string {
		return "";
	}
}

File: src/java/magma/app/compile/type/ArrayType.java
package magma.app.compile.type;

import magma.api.Type;

public record ArrayType(Type child) implements Type {
    @Override
    public String generate() {
        return this.child.generate() + "[]";
    }

    @Override
    public boolean isFunctional() {
        return false;
    }

    @Override
    public boolean isVar() {
        return false;
    }

    @Override
    public String generateBeforeName() {
        return "";
    }
}
